features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableCompoundButtonMapper.kt

```diff
@@ -36,27 +40,95 @@ internal abstract class CheckableCompoundButtonMapper<T : CompoundButton>(
+                checkableDrawableIndex
+            )
+        } else {
+            // view.buttonDrawable is not available below API 23, so reflection is used to retrieve it.
+            try {
+                mButtonDrawableField?.get(view) as? Drawable
+            } catch (e: IllegalAccessException) {
+                internalLogger.log(
+                    level = InternalLogger.Level.ERROR,
+                    targets = listOf(InternalLogger.Target.MAINTAINER, InternalLogger.Target.TELEMETRY),
+                    messageBuilder = { GET_DRAWABLE_FAIL_MESSAGE },
+                    throwable = e
+                )
+                null
+            } catch (e: IllegalArgumentException) {
+                internalLogger.log(
+                    level = InternalLogger.Level.ERROR,
+                    targets = listOf(InternalLogger.Target.MAINTAINER, InternalLogger.Target.TELEMETRY),
+                    messageBuilder = { GET_DRAWABLE_FAIL_MESSAGE },
+                    throwable = e
+                )
+                null
+            }
+        }
+        return originCheckableDrawable?.let { cloneCheckableDrawable(view, it) } ?: run {
+            internalLogger.log(
+                level = InternalLogger.Level.ERROR,
+                targets = listOf(InternalLogger.Target.MAINTAINER, InternalLogger.Target.TELEMETRY),
+                messageBuilder = { GET_DRAWABLE_FAIL_MESSAGE }
+            )
+            null
+        }
+    }
+
+    private fun cloneCheckableDrawable(view: T, drawable: Drawable): Drawable? {
+        return drawable.constantState?.newDrawable(view.resources)?.apply {
+            // Set state to make the drawable have correct tint.
+            setState(view.drawableState)
+            // Set tint list to drawable if the button has declared `buttonTint` attribute.
+            view.buttonTintList?.let {
+                setTintList(it)
+            }
+        }
+    }
+
     // endregion
 
     companion object {
-        internal const val MIN_PADDING_IN_PX = 20L
-        internal const val DEFAULT_CHECKABLE_HEIGHT_IN_PX = 84L
+        internal const val DEFAULT_CHECKABLE_HEIGHT_IN_DP = 32
+        internal const val GET_DRAWABLE_FAIL_MESSAGE =
+            "Failed to get buttonDrawable from the checkable compound button."
+
+        // Reflects the field at the initialization of the class instead of reflecting it for every wireframe generation
+        @Suppress("PrivateApi", "SwallowedException", "TooGenericExceptionCaught")
+        internal val mButtonDrawableField = try {
+            CompoundButton::class.java.getDeclaredField("mButtonDrawable").apply {
+                isAccessible = true
+            }
+        } catch (e: NoSuchFieldException) {
+            null
+        } catch (e: SecurityException) {
+            null
+        } catch (e: NullPointerException) {
+            null
+        }
```

Comment:
Don't suppress `SwallowedException`. In fact you can simply create a method which will get a value with 2 calls inside:

```
val drawableField = CompoundButton::class.java.getDeclaredField("mButtonDrawable").apply {
    isAccessible = true
}
return drawableField?.get(view) as? Drawable
```

this will allow you to have an isolated reflection usage site with a common handler for the exceptions coming from both calls and inject `internalLogger` as method argument to log exceptions.

Reply:
Actually this is a result of discussion with @xgouchet we prefer using field instead of a method because the reflection may can have some performance issue if we call it for every wireframe creation, so we just do it once during the initialisation of the class if we use a static field.

Reply:
makes sense indeed then